projects/vdk-plugins/vdk-ipython/src/vdk_ipython.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+import pathlib
+
+from IPython import get_ipython
+from IPython.core.magic_arguments import argument
+from IPython.core.magic_arguments import magic_arguments
+from IPython.core.magic_arguments import parse_argstring
+from vdk.internal.builtin_plugins.run.standalone_data_job import (
+    StandaloneDataJobFactory,
+)
+
+
+def load_ipython_extension(ipython):
+    """
+    IPython will look for this function specifically.
+    See https://ipython.readthedocs.io/en/stable/config/extensions/index.html
+    """
+    ipython.register_magic_function(magic_load_job, magic_name="reload_job_input")
+
+
+@magic_arguments()
+@argument("-p", "--path", type=str, default=None)
+@argument("-n", "--name", type=str, default=None)
+@argument("-a", "--arguments", type=str, default=None)
+@argument("-t", "--template", type=str, default=None)
+def magic_load_job(line: str):
+    """
+    You can use %initialize_vdk_job line magic within your Notebook to reload the job_input variable
+    for your current job
+    See more for line magic: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+    """
+    args = parse_argstring(magic_load_job, line)
+    load_job(args.path, args.name, args.arguments, args.template)
+
+
+def load_job(
+    path: str = None, name: str = None, arguments: str = None, template: str = None
+):
+    path = pathlib.Path(path) if path else pathlib.Path(os.getcwd())
+    with StandaloneDataJobFactory.create(
+        data_job_directory=path, name=name, job_args=arguments, template_name=template
+    ) as job_input:
+        get_ipython().push(variables={"job_input": job_input})
```

Comment:
The way it's written, we store "job_input" in get_ipython().push . and then load_job exits and that means that StandaloneDataJobFactory.__exit__ will be called and will finalize the job. 

OK. .. Let's merge it this way but this is not what we want, right?

Reply:
No, maybe we will need to push StandaloneDataJobFactory instead of this, which will give the user more responsibility, tho. I will try to fix this in further changes.

Reply:
is with resources closing the job_input after the code block containing push?

Reply:
not quite sure, since the documentation of push method is not much detailed. I am not sure whether it creates a copy of the job_input or not. This will be checked in the later changes, since it will take some time, it would be better to not include t in this PR.